test/e2e_cri/container.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_cri
+
+import (
+	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("Test CRI container", func() {
+	f := framework.NewDefaultCRIFramework("CRI-container-test")
+
+	var c internalapi.RuntimeService
+
+	BeforeEach(func() {
+		c = f.CRIClient.CRIRuntimeClient
+	})
+	Context("basic container operatings", func() {
+		var podID string
+		var podConfig *runtimeapi.PodSandboxConfig
+
+		BeforeEach(func() {
+			podID, podConfig = createPodSandboxForContainer(c)
+		})
+
+		AfterEach(func() {
+			By("stop PodSandbox")
+			c.StopPodSandbox(podID)
+			By("delete PodSandbox")
+			c.RemovePodSandbox(podID)
+		})
+
+		It("test basic operations on container", func() {
+			By("test create container")
+			containerID := testCreateContainer(c, podID, podConfig)
+
+			By("test list container")
+			containers := listContainerWithIDOrFail(c, containerID)
+			Expect(containerFound(containers, containerID)).To(BeTrue(), "container should be created")
+
+			By("test start container")
+			testStartContainer(c, containerID)
+
+			By("test stop container")
+			testStopContainer(c, containerID)
+
+			By("test remove container")
```

Comment:
This is one big test case, if ```By("test start container")``` fails it won't check if removing container works correctly.
It should be split to five independent test cases